base/i18n/icu_util.cc

```diff
@@ -86,15 +86,19 @@ void LazyInitIcuDataFile() {
   if (g_icudtl_pf != kInvalidPlatformFile) {
     return;
   }
-#if defined(OS_ANDROID)
-  int fd = base::android::OpenApkAsset(kAndroidAssetsIcuDataFileName,
-                                       &g_icudtl_region);
-  g_icudtl_pf = fd;
-  if (fd != -1) {
-    return;
-  }
-// For unit tests, data file is located on disk, so try there as a fallback.
-#endif  // defined(OS_ANDROID)
+//It's not correct to check and read from assets folder first, since xwalk
+//put icudtl.dat in other folders like res/raw instead of assets.Reading
```

Comment:
If icudtl.dat was moved to assets, it'll have coexistence issue with stock android Webview. Since there's only one assets folder for application, once app call android webview's function, it'll copy android webview's icudtl.dat to assets.  The two icudtl.dat will conflict.